server/routes/analyticsRouter.ts

```diff
@@ -51,8 +51,11 @@ export default function routes(router: Router): Router {
       { text: 'Protected characteristics' }
     )
 
-    const prisonersByEthnicity = { dataSource: 'NOMIS', lastUpdated: new Date() }
-    const prisonersInAgeGroups = { dataSource: 'NOMIS', lastUpdated: new Date() }
+    const activeCaseLoad = res.locals.user.activeCaseload.id
+
+    const analyticsService = new AnalyticsService(urlForLocation)
+    const prisonersByEthnicity = await analyticsService.getIncentiveLevelsByEthnicity(activeCaseLoad)
+    const prisonersInAgeGroups = await analyticsService.getIncentiveLevelsByAgeGroup(activeCaseLoad)
```

Comment:
Just a though so no need to change anything (especially given all is in flux), but it seems like these two could be done "concurrently" as the second call doesn't depend on the response of the first call.

Could be a good use case of `Promise.all`, e.g.

```javascript
const [prisonersByEthnicity, prisonersInAgeGroups] = await Promise.all([
    analyticsService.getIncentiveLevelsByEthnicity(activeCaseLoad),
    analyticsService.getIncentiveLevelsByAgeGroup(activeCaseLoad),
])
```

(or something like that - but as I said, just a thought, what you have is perfectly fine and clear)

Reply:
you're right if the 2 data sets come from different files in S3. however, i'm pretty sure that they will actually be from one table grouped by different columns so it won't make a difference as there will only be one async call to S3